batcher/aligned-batcher/src/lib.rs

```diff
@@ -261,17 +262,35 @@ impl Batcher {
     }
 
     pub async fn listen_new_blocks(self: Arc<Self>) -> Result<(), BatcherError> {
-        let mut stream = self
-            .eth_ws_provider
-            .subscribe_blocks()
-            .await
-            .map_err(|e| BatcherError::EthereumSubscriptionError(e.to_string()))?;
+        retry_function(
+            || {
+                let app = self.clone();
+                async move { app.listen_new_blocks_retryable().await }
```

Comment:
Does this work? Isn't this kind of like having an actual async lambda?

Reply:
I tested it and it worked for me.